File: public/examples/integrations/mcp-servers/hubspot_crm_api/get_goal_targets_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "HubspotCrmApi.GetGoalTargets"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'associated_object_types': ['contact', 'company'],
    'paging_cursor_token': 'abc123',
    'properties_with_history': ['status', 'due_date'],
    'results_per_page_limit': 10,
    'return_only_archived': False,
    'returned_properties': ['name', 'goal_value']
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/hubspot_crm_api/get_goal_targets_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "HubspotCrmApi.GetGoalTargets";

// Start the authorization process
const authResponse = await client.tools.authorize({
  tool_name: TOOL_NAME,
  user_id: USER_ID,
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "associated_object_types": [
    "contact",
    "company"
  ],
  "paging_cursor_token": "abc123",
  "properties_with_history": [
    "status",
    "priority"
  ],
  "results_per_page_limit": 10,
  "return_only_archived": false,
  "returned_properties": [
    "name",
    "due_date"
  ]
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
